src/firejail/main.c

```diff
@@ -1773,6 +1774,14 @@ int main(int argc, char **argv) {
 			else
 				exit_err_feature("private-cache");
 		}
+		else if (strcmp(argv[i], "--private-cwd") == 0) {
+			cfg.cwd = NULL;
+			arg_private_cwd = 1;
+		}
+		else if (strncmp(argv[i], "--private-cwd=", 14) == 0) {
+			cfg.cwd = argv[i] + 14;
```

Comment:
Can you check the argument with the `invalid_filename()` function against weird characters?

Though not strictly necessary, I also wonder if the the previous cfg.cwd should be `free`'d and the new cfg.cwd should be allocated. Just for keeping it consistent.